### PR TITLE
[Filebeat] NetFlow input support for custom field definitions

### PR DESCRIPTION
**Pull request by adriansr**
This PR adds support for loading custom (enterprise-specific) fields to the Filebeat NetFlow input. 
These fields can extend and/or override fields in NetFlow V9 and IPFIX. 

For compatibility, the feature uses the same [field definition YAML format](https://www.elastic.co/guide/en/logstash/current/plugins-codecs-netflow.html#plugins-codecs-netflow-ipfix_definitions) as Logstash's netflow codec plugin.

A new configuration option `custom_definitions` consists of a list of paths to definition files.

### CODE REVIEW

File: x-pack/filebeat/input/netflow/definitions.go [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+		}
+		if err := loadFields(tree, uint32(pem), defs); err != nil {
+			return nil, errors.Wrapf(err, "failed to load IPFIX fields for pem=%d", pem)
+		}
+	}
+	return defs, nil
+}
+
+// LoadFieldDefinitionsFromFile takes the path to a YAML file in Logstash
+// Netflow or IPFIX custom fields format and converts it to a FieldDict.
```
Comment (andrewkroh) [inline x-pack/filebeat/input/netflow/definitions.go]: We'll need a description of the config format in the Filebeat docs.

Comment (adriansr) [reply x-pack/filebeat/input/netflow/definitions.go]: Done

File: x-pack/filebeat/input/netflow/netflow_test.go [hunk 2]
```diff
@@ -148,7 +154,7 @@ func readDatTests(t testing.TB) *DatTests {
 	return &tests
 }
 
-func getFlowsFromDat(t testing.TB, name string, datFiles ...string) TestResult {
+func getFlowsFromDat(t testing.TB, name string, testData DatTest) TestResult {
```
Comment (webmat) [inline x-pack/filebeat/input/netflow/netflow_test.go]: Nit: This mixed naming in var/funcs named `dat` vs `data` is confusing. Both are right, but I think if you pick one or the other across the board, things will be more clear. I think "data" better describes it.

File: x-pack/filebeat/input/netflow/decoder/fields/field_test.go [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+		t.FailNow()
+	}
+
+	f.Merge(c)
+	assert.Len(t, f, len(a)+len(b)+len(c)-1)
+	if !checkContains(t, f, c) {
+		t.FailNow()
+	}
+
+}
```
[the remark this answers is not in the document]
Comment (adriansr) [reply x-pack/filebeat/input/netflow/decoder/fields/field_test.go]: There's a conflict with `Key{3, 4}`, which is overwritten by c. Is that what you mean?

**Pull request merged**